tests/Integration/Behaviour/Features/Context/Domain/OrderMessageContext.php

```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * 2007-2019 PrestaShop and Contributors
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2019 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace Tests\Integration\Behaviour\Features\Context\Domain;
+
+use OrderMessage;
+use PrestaShop\PrestaShop\Core\Domain\OrderMessage\Command\AddOrderMessageCommand;
+use PrestaShop\PrestaShop\Core\Domain\OrderMessage\ValueObject\OrderMessageId;
+
+class OrderMessageContext extends AbstractDomainFeatureContext
+{
+    /**
+     * @When I specify :propertyName :propertyValue in default language for order message :reference
+     */
+    public function specifyPropertyInDefaultLanguage(string $propertyName, string $propertyValue, string $reference)
```

Comment:
I'm not a fan 😐 of these functions because you "stack" property specifications until you perform the persistence call using `addWithSpecifiedProperties`. This can lead to debugging issues when `addWithSpecifiedProperties` is called too soon, or called by another function (without you aware of it) or called too late and leading to incomplete / invalid data set.

You really dont want to use the `TableNode` provided by Behat 😛 ?
```
    When I add order message "OM1" with specified properties for default language:
 | name       | Out of stock                       |
 | message | Products are out of stock |
```

Reply:
Note : this is not a blocking item, I'm not going to bother you too much on behat tests. Adding integration tests is already one hundred times better than migrating pages without them 😛 so if you tell me you really feel like if it's not an issue, I'm fine 👍

Reply:
Well, as I said before, for me it seems like more readable format. For example, imagine having this
```php
| active | true |
```

instead of this

```php
And I specify "Displayed" is "enabled"
```

But if you feel it's better to use tables, I can do that in next PRs. :)

Reply:
Well, as I said before, for me it seems like more readable format. For example, imagine having this
```php
| active | true |
```

instead of this

```php
And I specify "Displayed" is "enabled"
```

But if you feel it's better to use tables, I can do that in next PRs. :)

Reply:
Well, as I said before, for me it seems like more readable format. For example, imagine having this
```php
| active | true |
```

instead of this

```php
And I specify "Displayed" is "enabled"
```

But if you feel it's better to use tables, I can do that in next PRs. :)